consensus/core/src/dag_state.rs

```diff
@@ -257,6 +261,86 @@ impl DagState {
         Ok(blocks)
     }
 
+    /// Retrieves the last block proposed for the specified `authority`. If no block is found in cache
+    /// then the genesis block is returned as no other block has been received from that authority.
+    pub(crate) fn get_last_block_for_authority(&self, authority: AuthorityIndex) -> VerifiedBlock {
+        if let Some(last) = self.cached_refs[authority].last() {
+            return self
+                .recent_blocks
+                .get(last)
+                .expect("Block should be found in recent blocks")
+                .clone();
+        }
+
+        // if none exists, then fallback to genesis
+        let (_, genesis_block) = self
```

Comment:
I believe `self.genesis` can be used.

Reply:
`self.genesis` is being used, what am I missing?